File: WarStory/WarStory/Scripts/domain/Warehouse.js
var Warehouse = function (config) {
    this.name = config.name;
    this.wood = parseInt(config.wood);
    this.stone = parseInt(config.stone);
    this.iron = parseInt(config.iron);

    this.forestLv = parseInt(config.forestLv);
    this.mineLv = parseInt(config.mineLv);
    this.mountainLv = parseInt(config.mountainLv);
}

Warehouse.prototype.action = function () {
    //console.log(this.name + ".action");

    this.wood = this.wood + this.forestLv;
    this.stone = this.stone + this.mountainLv;
    this.iron = this.iron + this.mineLv;
}

File: WarStory/WarStoryServer/Domain/Warehouse.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

using MongoDB.Bson.Serialization.Attributes;

namespace WarStoryServer.Domain
{
    public class Warehouse : Actor
    {
        [BsonId]
        public string Name { get; set; }

        public int Wood { get; set; }
        public int Stone { get; set; }
        public int Iron { get; set; }

        public override void Action()
        {
            ++Wood;
            ++Stone;
            ++Iron;
        }
    }
}
